xgcm/autogenerate.py

```diff
@@ -0,0 +1,274 @@
+from __future__ import print_function
+from future.utils import iteritems
+from .grid import Axis
+
+
+def generate_axis(ds,
+                  axis,
+                  name,
+                  axis_dim,
+                  pos_from='center',
+                  pos_to='left',
+                  wrap=None,
+                  pad=None,
+                  new_name=None,
+                  raw_switch=True):
+    """
+    Creates c-grid dimensions (or coordinates) along an axis of
+    Parameters
+    ----------
+    ds : xarray.Dataset
+        Dataset with gridinformation used to construct c-grid
+    axis : str
+        The appropriate xgcm axis. E.g. 'X' for longitudes.
+    name : str
+        The name of the variable in ds, providing the original grid.
+    axis_dim : str
+        The dimension of ds[name] corresponding to axis. If name itself is a
+        dimension, this should be equal to name.
+    pos_from : {'center','left','right'}
+        Position of the gridpoints given in 'ds'.
+    pos_to : {'center','left','right'}
+        Position of the gridpoints to be generated.
+    wrap : None or float
+        If specified, marks the value of discontinuity across boundary, e.g.
+        360 for global longitude values and 180 for global latitudes.
+    pad : {None, float, 'auto'}
+        If specified, determines the padding to be applied across boundary.
+        If float is specified, that value is used as padding. Auto attempts to
+        pad linearly extrapolated values. Can be useful for e.g. depth
+        coordinates (to reconstruct 0 depth). Can lead to unexpected values
+        when coordinate is multidimensional.
+    new_name : str
+        Name of the inferred grid variable. Defaults to name+'_inferred'
+    raw_switch : bool
+        Determines if the attributes are created from scratch. Should be
+        enabled for dimensions and deactivated for multidimensional
+        coordinates. These can only be calculated after the dims are created.
```

Comment:
I would still prefer a different name for `raw_switch`. Why not `create_attributes_from_scratch`? Something more verbose that corresponds with what the option actually does.